configure

```diff
@@ -5333,13 +5263,29 @@ EOF
     fi
     rm -f conftest*
     ;;
-  *)
-    with_dfp=no ;;
   esac
 
 fi
 
 
+# Test if the host as configured by the environment supports hard-dfp
+# by scanning assembler for soft-dfp
+if test -z "$with_dfp"; then
+  gcc_intrin=`echo "_Decimal128 x(_Decimal64 y) { return y; }" \
+              | ${CC-cc} ${CFLAGS} -S -c -xc - -o - \
+              | grep extendddtd -m1`
+  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: gcc_intrin is ${gcc_intrin}" >&5
+$as_echo "$as_me: WARNING: gcc_intrin is ${gcc_intrin}" >&2;}
+  case "$gcc_intrin" in
+    *extendddtd*)
+      with_dfp=no
+      ;;
+    *)
+      with_dfp=yes
+      ;;
+  esac
+fi
+
 if test -n "$submachine"; then
```

Comment:
Will this ever be not null?

Reply:
No, with hindsight, it should be set by --with-cpu to preserve the old behavior of setting -mcpu/-march.  Thanks.